 ## AVIATION RISK ANALYSIS

 ## INTRODUCTION
The company wants to start working with airplanes but needs to know which ones are the safest.  

In this project, I will look at accident data from 1962–2023.  
The goal is to find the lowest-risk aircraft and give 3 clear recommendations to help the business decide which planes to buy.


 ## BUSINESS UNDERSTANDING
 
 ## WHO'S THIS FOR?  
The Head of the Aviation Division.  

 ## WHAT DOES THE BUSINESS NEED? 
They want to know which aircraft are safest to buy.  

Key Questions:  
1. Which aircraft types have the fewest accidents?  
2. How have accidents changed over time?  
3. Are private or commercial aircraft safer?  


## DATA UNDERSTANDING

The dataset comes from the National Transportation Safety Board (NTSB).  
It covers aviation accidents from 1962 to 2023.  

 ## WHAT WILL I DO HERE:
- Load the dataset into Python.  
- Look at the first few rows to see what it looks like.  
- Check the number of rows, columns, and missing values.  
- Find the most important columns for our analysis (like aircraft type, purpose, fatalities, and damage).  


## LOAD DATASET AND INSPECT AND UNDERSTAND THE DATA
Here we will load the dataset into a DataFrame,  
look at the first rows, and check the shape and basic info.


In [111]:
import pandas as pd

# load the dataset and assign it to tb 
tb = pd.read_csv('aviation-accident-data-2023-05-16.csv')

# print the first 5 rows of the aviation dataset 
tb.head()

date             type registration       operator fatalities  \
0  date unk.   Antonov An-12B       T-1206  Indonesian AF        NaN   
1  date unk.   Antonov An-12B       T-1204  Indonesian AF        NaN   
2  date unk.   Antonov An-12B       T-1201  Indonesian AF        NaN   
3  date unk.  Antonov An-12BK          NaN      Soviet AF        NaN   
4  date unk.  Antonov An-12BP   CCCP-11815      Soviet AF          0   

              location          country cat     year  
0                  NaN  Unknown country  U1  unknown  
1                  NaN  Unknown country  U1  unknown  
2                  NaN  Unknown country  U1  unknown  
3  Tiksi Airport (IKS)           Russia  A1  unknown  
4  Massawa Airport ...          Eritrea  A1  unknown

In [112]:
# let's check the shape of the dataset (rows,columns)
tb.shape
# we can confirm that we have 23967 rows and 9 columns in the  aviation dataset

(23967, 9)

In [113]:
# let's check the data types and column names
tb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23967 entries, 0 to 23966
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   date          23967 non-null  object
 1   type          23967 non-null  object
 2   registration  22419 non-null  object
 3   operator      23963 non-null  object
 4   fatalities    20029 non-null  object
 5   location      23019 non-null  object
 6   country       23967 non-null  object
 7   cat           23967 non-null  object
 8   year          23967 non-null  object
dtypes: object(9)
memory usage: 1.6+ MB


In [114]:
# we can confirm that we have 9 colums with 23967 rows but some columns have missing values or duplicate values thus leading to the uneven number in some columns

## DATA CLEANING

Now we will check for missing values and clean the data.  
Steps:
- Count missing values in each column  
- Decide which columns to drop or fix  
- Rename columns for easy readability and understanding  


In [115]:
# we'll start by counting missing values
tb.isnull().sum()

date               0
type               0
registration    1548
operator           4
fatalities      3938
location         948
country            0
cat                0
year               0
dtype: int64

we can identify that 4 columns have missing values but for easier readability we'll arrange them in descending order

In [116]:
# arrange them in descending order
tb.isnull().sum().sort_values(ascending=False)

fatalities      3938
registration    1548
location         948
operator           4
date               0
type               0
country            0
cat                0
year               0
dtype: int64

In [117]:
# now let's check for duplicate values
tb.duplicated().sum()

np.int64(115)

we can confirm that we have 115 duplicates in our aviation dataset now we'll rename the columns for easier understanding and readability

## RENAME COLUMNS

The current column names are confusing and not detailed .  
We will rename them to be easier to read and clean and analyze.


In [118]:
# renaming the columns too be user friendly and easily understandable

tb = tb.rename(columns={
     "date": "Date of Accident",
    "type": "Aircraft Type",
    "registration": "Aircraft Registration",
    "operator": "Operator",
    "fatalities": "Deaths",
    "location": "Location",
    "country": "Country",
    "cat": "Category",
    "year": "Year"
})

In [119]:
# let's check whether the columns changed 

tb.head(5)

Date of Accident    Aircraft Type Aircraft Registration       Operator  \
0        date unk.   Antonov An-12B                T-1206  Indonesian AF   
1        date unk.   Antonov An-12B                T-1204  Indonesian AF   
2        date unk.   Antonov An-12B                T-1201  Indonesian AF   
3        date unk.  Antonov An-12BK                   NaN      Soviet AF   
4        date unk.  Antonov An-12BP            CCCP-11815      Soviet AF   

  Deaths             Location          Country Category     Year  
0    NaN                  NaN  Unknown country       U1  unknown  
1    NaN                  NaN  Unknown country       U1  unknown  
2    NaN                  NaN  Unknown country       U1  unknown  
3    NaN  Tiksi Airport (IKS)           Russia       A1  unknown  
4      0  Massawa Airport ...          Eritrea       A1  unknown

## Clean and Change Data Types and  Replace Missing Values

We will:
- Change Year and Fatalities to numbers  
- Replace missing values in Registration, Operator, and Location with "Unknown"  
- Replace missing Fatalities with 0  


In [120]:
# convert data types first
tb['Year'] = pd.to_numeric(tb['Year'], errors='coerce')
tb['Deaths'] = pd.to_numeric(tb['Deaths'], errors='coerce').fillna(0)



In [121]:
# Now let's fill in the missing values with data using fillna() method

tb['Location']=tb['Location'].fillna('Not Reported')
tb['Operator']=tb['Operator'].fillna('Not Reported')
tb['Aircraft Registration']=tb['Aircraft Registration'].fillna('Not Reported')




In [122]:
# let's check whether the missing values are filled
tb.isnull().sum()

Date of Accident          0
Aircraft Type             0
Aircraft Registration     0
Operator                  0
Deaths                    0
Location                  0
Country                   0
Category                  0
Year                     61
dtype: int64

### Handling Missing Data

Some columns had missing values (like Operator,Aircraft Registration, and Location).  
Instead of leaving them empty, we replaced them with **"Not Reported"**.  
This makes the dataset easier to read,analyze and avoids problems when  plotting data.  

For the **Deaths** column, missing values were filled with **0** since no number was reported.
